pkg/kubelet/kubelet.go

```diff
@@ -880,6 +880,10 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string) (map[string]string, error) {
 
 	// project the services in namespace ns onto the master services
 	for _, service := range services.Items {
+		// ignore services where PortalIP is "None"
+		if service.Spec.PortalIP == "None" {
```

Comment:
Seems like we should skip services with no PortalIP (`""`) also. I don't see that being done in FromServices().